src/components/sidebar/sidebar.component.js

```diff
@@ -9,6 +9,8 @@ import FocusTrap from "../../__internal__/focus-trap";
 import SidebarHeader from "./__internal__/sidebar-header";
 import Box from "../box";
 
+export const SidebarContext = React.createContext();
```

Comment:
This needs a default value 

`export const SidebarContext = React.createContext({});`

Reply:
Good spot!